### PR TITLE
[DEVHAS-102] Add KCP-specific manifests and docs

### PR DESCRIPTION
**Pull request by johnmcollier**
Related to: https://issues.redhat.com/browse/DEVHAS-102

This PR adds the following manifests for running HAS on KCP:

- `config/kcp` 
  - Deploys HAS on KCP without webhooks.
- `config/kcp-webhooks`
  - Enables webhooks for HAS on KCP
  - **Note:** Enabling webhooks is still a bit of a manual process even with this Kustomize manifest, and the `kcp-webhooks.md` doc that I am adding **must** be followed when enabling webhooks
- `config/kcp-rbac`
  - A copy of `config/rbac` with certain manifests removed that were not needed for KCP
  

The following docs have been modified or added
- `docs/kcp.md`
    - Covers how to run KCP on a managed instance of KCP (e.g. kcp-stable or kcp-unstable)
- `docs/kcp-webhooks.md`
     - Covers how to enable webhooks for HAS, _after_ installing HAS on KCP
- `docs/kcp-old.md` (formerly `kcp.md`) 
    - The now former kcp doc. Instead of removing it, I opted to keep it, as it contains useful information on running HAS on a **local** instance of KCP

**How to Test?**
1) Get yourself access to a managed KCP instance (as of right now, `kcp-unstable` must be used)
2) Checkout my changes
3) Follow the instructions in `./docs/kcp.md` to install HAS on KCP
   - After deploy on kcp-unstable, HAS may be stuck crashlooping. If so, do the following:
       - `kubectl edit deploy <deploy> -n $KCP_NAMESPACE`
       - Remove the following from the deployment:
          ```
          - configMap:
              items:
              - key: ca.crt
                path: ca.crt
              name: kcp-root-ca.crt
          ```

### CODE REVIEW

File: docs/kcp-webhooks.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  
+    - *NOTE:* You must make sure the generated url for the Route is less than 63 characters, or else OpenShift Acme's cert generation *will* fail. 
+  
+    - If the generated route url is over 63 characters, modify it to be shorter.
+
+6) Access the Route URL in your browser and verify that publicly signed certificates are being used.
+
+Next, switch back to KCP and do the following
+
+7) Open `config/kcp-webhook/manifests.yaml` in an editor and change all occurrences of `<REPLACE_ME>` to the exposed Route URL from the previous step
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply docs/kcp-webhooks.md:77]: Updated.

File: docs/kcp-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+2) `oc apply -f hack/kcp/webhook-service.yaml -n $KCP_NAMESPACE` to create the Service exposing the webhook service
+
+3) A secret, `webhook-server-cert` should have been created in the same namespace. Run `oc extract secret/webhook-server-cert -n $KCP_NAMESPACE` to retrieve the TLS certificates for the webhook service.
+
+4) Copy the contents of the `tls.crt` file.
+
+5) Open `hack/kcp/webhook-route.yaml` in an editor and make the following changes: 
+  - Replace the `<REPLACE_CERT>` with the TLS Certificate retrieved from the previous step. 
+  - Replace the `<REPLACE_HOST>` with a valid Route hostname (e.g. webhook.apps.myopenshiftcluster.com). We are not using generated hostnames because the generated hostname is too long (> 63 characters) because of the kcp namespace length.
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply docs/kcp-webhooks.md:35]: Updated.

**Comment by johnmcollier** [comment]
@maysunfaisal Review comments addressed.

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**